micronaut-camunda-bpm-feature/src/test/kotlin/info/novatec/micronaut/camunda/bpm/feature/test/MnProcessEngineConfigurationTest.kt

```diff
@@ -0,0 +1,114 @@
+        // process has been finished but rollback happened and we cannot find it in the history
+        assertEquals(0, findHistoricProcessInstances(WITH_ROLLBACK).size)
+    }
+
+    @Test
+    open fun testSurroundingTransactionWithCommit() {
+        transactionManager.executeWrite { transactionStatus: TransactionStatus<Connection?> ->
+            try {
+                startProcess(TX_WITH_COMMIT)
+                return@executeWrite startProcess(TX_WITH_COMMIT)
+            } finally {
+                assertFalse(transactionStatus.isRollbackOnly)
+            }
+        }
+        // both processes have been finished with commit and we can find them in the history
+        assertEquals(2, findHistoricProcessInstances(TX_WITH_COMMIT).size)
+    }
+
+    @Test
+    open fun testSurroundingTransactionWithRollback() {
+        assertThrows(RuntimeException::class.java) {
+            transactionManager.executeWrite(TransactionCallback<Connection, String> { transactionStatus: TransactionStatus<Connection> ->
+                try {
+                    startProcess(TX_WITH_ROLLBACK)
+                    return@TransactionCallback startProcessWithRuntimeError(TX_WITH_ROLLBACK)
+                } finally {
+                    assertTrue(transactionStatus.isRollbackOnly)
+                }
+            })
+        }
+        // first process has finished successfully but rollback happened and we cannot find anything in the history
+        assertEquals(0, findHistoricProcessInstances(TX_WITH_ROLLBACK).size)
+    }
+
+    @Test
+    open fun testNoSurroundingTransactionWithCommits() {
+        startProcess(WITH_COMMIT_COMMIT)
+        startProcess(WITH_COMMIT_COMMIT)
+        // both processes have been finished with commit and we can find them in the history
+        assertEquals(2, findHistoricProcessInstances(WITH_COMMIT_COMMIT).size)
+    }
+
+    @Test
+    open fun testNoSurroundingTransactionWithCommitAndRollback() {
+        startProcess(WITH_COMMIT_ROLLBACK)
+        assertThrows(RuntimeException::class.java) { startProcessWithRuntimeError(WITH_COMMIT_ROLLBACK) }
+        // first process has finished successfully but rollback happened and we can find only one in the history
+        assertEquals(1, findHistoricProcessInstances(WITH_COMMIT_ROLLBACK).size)
+    }
+
+    open fun startProcess(businessKey: String): String {
+        return runtimeService.startProcessInstanceByKey("ProcessEmpty", businessKey).id
+    }
+
+    fun startProcessWithRuntimeError(businessKey: String): String {
+        return runtimeService.startProcessInstanceByKey("ProcessRuntimeError", businessKey).id
+    }
+
+    fun findHistoricProcessInstances(businessKey: String): List<HistoricProcessInstance> {
+        return historyService.createHistoricProcessInstanceQuery()
+                .processInstanceBusinessKey(businessKey)
+                .list()
+    }
+
+    companion object {
+        const val WITH_COMMIT = "withCommit"
```

Comment:
same as I mentioned above.

Reply:
this is also referenced in a test. So I guess this should also stay, right?

btw, the companion object was created automatically by IDEA when converting Java to Kotlin.

Reply:
Then it should stay this way :+1: